text/0021-internal-data-monitoring.md

```diff
@@ -0,0 +1,182 @@
+---
+rfc pr: https://github.com/UCLA-Creative-Labs/cl-rfcs/pull/22
+tracking issue: https://github.com/UCLA-Creative-Labs/cl-rfcs/issues/21
+---
+
+# Watchdog
+
+`Watchdog` is a monorepo that will contain a variety of Creative Labs
+data analysis tools for internal use. 
+
+## Blurb
+
+The goal of this project is to allow all the teams across Creative Labs
+board to make data-driven decisions. We want to set up a system where all
+members can access previous data and use insights from the past to make
+good judgement calls in the present.
+
+Watchdog is meant to be a monorepo to house various data analysis tools
+for Creative Labs. There are a few ideas to implement currently. These
+projects are primarily meant for Creative Labs board members, but we
+could look towards displaying some of the data publicly or publishing
+our findings through a blog post for marketing content!
+
+Note: Since we do not have complete datasets on many events or projects in
+the past, much of this will be just to setup infrastructure for the future.
+
+### Project 1: Project Data Vis
+The first project is a continuation of the current Watchdog project, which
+is a tool to get data analytics on Creative Labs projects. Currently, it
+fetches project application information and displays its various statistics.
+We want to organize this by creating user datasets that would keep track
+of the same applicant through multiple application cycles. There would be a
+web app that displays the most important datasets in a visualized format and
+key conclusions would be highlighted. An addition would be to allow board
+members to better access the data by allowing users to choose their own
+method of displaying data (bar graph, pie chart, etc) and choosing a
+specific time period with various filters.
+
+### Project 2: Social Media Data Vis
```

Comment:
I would also like you to expand on this section further. The entire point of RFC is not to just record your idea! You want to layout how you plan on implementing your solution so that you know how to move forward when this is approved!

This RFC should give you enough context about your entire project and how it will come together. Currently, I have very little idea where this project will go.

Questions to ask yourself:
* How are you going to pair social media posts to an event?
* What kind of api's are you going to use to get this? And is it even possible with those APIs
* Is this going to be a function or a server? How will this be triggered? Where. arewe going to store all the information?

Reply:
i dont believe this is resolved